### PR TITLE
Added progress report in algorithms: CreatePeaksWorkspace, CreatePSDBleedMask

### LINKED ISSUE #14294: Add progress report in algorithms: CreatePeaksWorkspace, CreatePSDBleedMask, DeleteWorkspace, FFT
These algorithm don't seem to report progress.

There are some examples in the usage examples / doc tests that could be used to work on this ticket/issue.

For an example of algorithm that does report progress status, see SumSpectra or Rebin. I think that those two are good examples.

This may be helpful:

C++ API doxygen page: http://doxygen.mantidproject.org/nightly/de/dd6/classMantid_1_1API_1_1Progress.html
Python: http://www.mantidproject.org/PythonAlgorithm_Progress_Reporting

### PR DESCRIPTION
**Pull request by LamarMoore**
Fixes #14294 

The release notes have not been updated.
## Changes

The algorithms CreatePeaksWorkspace and CreatePSDBleedMask are now reporting progress. The other algorithms listed in the original issue execute very quickly and therefore do not need progress reports.
### CreatePeaksWorkspace

Test in script editor:

_sampleWs = CreateSampleWorkspace(BankPixelWidth=50, NumBanks=100)_
_ws = CreatePeaksWorkspace(InstrumentWorkspace=sampleWs,NumberOfPeaks=3)_
_print "Created a %s with %i rows" % (ws.id(), ws.rowCount())_
### CreatePSDBleedMask

Test in script editor:

_import numpy as np_

_ws=CreateSampleWorkspace(BankPixelWidth=20, NumBanks=100)_
_AddSampleLog(ws,"goodfrm","10","Number")_
_noisyY =  np.array(ws.readY(0))_
_noisyY[0]=1e20_
_ws.setY(50,noisyY)_
_(wsOut, numFailures) = CreatePSDBleedMask(ws,MaxTubeFramerate=10,_ _NIgnoredCentralPixels=2)_

_print "%i spectra have been masked in wsOut" % numFailures_

### CODE REVIEW

File: Framework/Algorithms/src/CreatePSDBleedMask.cpp [hunk 1]
```diff
@@ -30,8 +30,7 @@ using DataObjects::MaskWorkspace_sptr;
 //----------------------------------------------------------------------
 
 /// Default constructor
-CreatePSDBleedMask::CreatePSDBleedMask()
-    : m_maxRate(0.0), m_numIgnoredPixels(0), m_isRawCounts(false) {}
+CreatePSDBleedMask::CreatePSDBleedMask() {}
```
Comment (peterfpeterson) [inline Framework/Algorithms/src/CreatePSDBleedMask.cpp]: This change will likely create coverity warnings/errors. Initializer lists are a good way of setting default values.

**Comment by peterfpeterson** [comment]
Please put back the member variable initialization in the default constructor of `CreatePSDBleedMask`. Coverity is going to complain in a couple days after it is merged into master if you don't.

**Comment by LamarMoore** [comment]
The member variables have been removed. There is no need to initialize them.

**Comment by peterfpeterson** [comment]
I didn't notice that.

**Pull request merged**